app/src/main/java/uk/ac/bath/petmatch/PetProfileActivity.java

```diff
@@ -62,6 +63,17 @@ public void onClick(View v) {
 
 
         final Button button = findViewById(R.id.editButton);
+        if (loginService.getLoggedInUser() != null){
+            if (loginService.getLoggedInUser().getShelter() !=null){
```

Comment:
This is good, but only half way there! You were also supposed to verify, that the pet belongs in the user's shelter!